tests/systemd_testsuite/prepare_systemd_and_testsuite.pm

```diff
@@ -6,20 +6,90 @@
 # Summary: Prepare systemd and testsuite.
 # Maintainer: Sergio Lindo Mansilla <slindomansilla@suse.com>, Thomas Blume <tblume@suse.com>
 
-use base 'systemd_testsuite_test';
-use warnings;
-use strict;
+use Mojo::Base qw(systemd_testsuite_test);
 use testapi;
+use utils;
+use version_utils qw(is_sle);
+use registration qw(add_suseconnect_product);
 
 sub run {
     my ($self) = @_;
-    $self->testsuiteinstall;
-    assert_script_run('cd /usr/lib/systemd/tests');
-    assert_script_run('./run-tests.sh --prepare', 600);
+    my $cgroups = get_var('SYSTEMD_UNIFIED_CGROUP', 'yes');
+    my $nspawn = get_var('SYSTEMD_NSPAWN', 1);
+    my $no_build = get_var('SYSTEMD_NO_BUILD', 1);
+    my $test_opts = {
+        NO_BUILD => $no_build,
+        TEST_PREFER_NSPAWN => $nspawn,
+        UNIFIED_CGROUP_HIERARCHY => $cgroups
+    };
+    my $testdir = '/usr/lib/systemd/tests/test/';
+    my @pkgs = qw(
+      lz4
+      busybox
+      qemu
+      dhcp-client
+      python3
+      plymouth
+      binutils
+      netcat-openbsd
+      cryptsetup
+      less
+      device-mapper
+      strace
+      e2fsprogs
+      hostname
+      net-tools-deprecated
+      systemd-testsuite
+    );
+
+    $self->select_serial_terminal();
+
+    if (is_sle) {
+        add_suseconnect_product('sle-module-legacy');
+        add_suseconnect_product('sle-module-desktop-applications');
+        add_suseconnect_product('sle-module-development-tools');
+        my $repo = sprintf('http://download.suse.de/download/ibs/SUSE:/SLE-%s:/GA/standard/',
+            get_var('VERSION'));
+        zypper_call("ar $repo systemd-tests");
+    }
+
+    #install testsuite and dependecies
+    zypper_call('ref');
+    zypper_call("in @pkgs");
+
+    # navigate to test case directory
+    # extract all available test cases
+    assert_script_run("cd $testdir");
+    my @tests = split(/\n/, script_output(qq(find . -maxdepth 1 -type d -name "TEST-*")));
+    my @schedule = ();
+    my $exclude = get_var('SYSTEMD_EXCLUDE');
+
+    foreach my $test (@tests) {
+        # trim folder prefix
+        $test =~ s/\.\///;
+        if (defined($exclude) && $test =~ m/$exclude/) {
+            next;
+        }
+        push @schedule, $test;
+    }
+
+    if (my $include = get_var('SYSTEMD_INCLUDE')) {
+        @schedule = ();
+        foreach my $test (split(',', $include)) {
+            push @schedule, $test;
+        }
```

Comment:
The for loop can be replaced with a single line:

    @schedule = split(',', $include);

and if `SYSTEMD_INCLUDE` is defined, you should skip scanning the SUT directory altogether.

Reply:
Oh, good point! Thanks!